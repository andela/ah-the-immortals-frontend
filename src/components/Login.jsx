import React, { Component } from 'react';
import { Container, Col, Row, Button } from 'react-bootstrap';
import LoginForm from './loginForm';
import LoginInformation from './loginInfo';

class Login extends Component {
  render() {
    return (
      <Row>
        <Col className="left-container">
          <Container>
            <LoginInformation />
          </Container>
        </Col>
        <Col className="right-container">
          <Container>
            <div className="top">
              <p className="form-title">Sign In</p>
              <LoginForm />
            </div>
          </Container>
        </Col>
      </Row>
    );
  }
}

export default Login;
