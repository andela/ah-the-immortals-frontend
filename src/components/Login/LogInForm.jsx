import React from 'react';
import { Link } from 'react-router-dom';
import PropTypes from 'prop-types';
import { Button } from 'react-bootstrap';

const LogInForm = ({ handleSignInSubmit, handleChange }) => (
  <form onSubmit={handleSignInSubmit} className="modal-form">
    <label htmlFor="email">
      <input type="email" required className="form-control" name="email" onChange={handleChange} placeholder="Email" />
    </label>
    <label htmlFor="password">
      <input type="password" required className="form-control" name="password" onChange={handleChange} placeholder="Password" />
    </label>
    <Button type="submit" variant="info" className="form-control btn" value="SIGN IN">
        SIGN IN
    </Button>
    <div className="reset-div">
      <Link to="/reset" className="passreset-link">Forgot Password?</Link>
    </div>
  </form>
);

LogInForm.propTypes = {
  handleSignInSubmit: PropTypes.func.isRequired,
  handleChange: PropTypes.func.isRequired,
};

export default LogInForm;
