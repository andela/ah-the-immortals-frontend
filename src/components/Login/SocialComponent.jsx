import React from 'react';
import PropTypes from 'prop-types';

const SocialComponent = ({ facebook, google, twitter }) => {
  return (
    <div>
      <button className="btn btn-link" id="fb" onClick={facebook} type="button">
        <img src="https://img.icons8.com/color/48/000000/facebook-circled.png" alt="avatar" />
      </button>
      <button className="btn btn-link" id="ggl" onClick={google} type="button">
        <img src="https://img.icons8.com/color/48/000000/google-plus.png" alt="avatar" />
      </button>
      <button className="btn btn-link" id="twt" onClick={twitter} type="button">
        <img src="https://img.icons8.com/color/48/000000/twitter-circled.png" alt="avatar" />
      </button>
    </div>
  );
};

SocialComponent.propTypes = {
  facebook: PropTypes.func.isRequired,
  google: PropTypes.func.isRequired,
  twitter: PropTypes.func.isRequired,
};

export default SocialComponent;
