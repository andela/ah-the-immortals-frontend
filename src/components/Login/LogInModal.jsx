import React from 'react';
import { Modal } from 'react-bootstrap';
import PropTypes from 'prop-types';
import '../../styles/style.css';
import '../../styles/App.css';
import LogInForm from './LogInForm';
import SocialComponent from './SocialComponent';


const LogInModal = ({
  closeModal,
  signInShow,
  handleChange,
  handleSignInSubmit,
  facebook,
  twitter,
  google
}) => {
  return (
    <Modal show={signInShow} onHide={closeModal}>
      <Modal.Header closeButton>
      </Modal.Header>
      <Modal.Body className="modal-body">
        <div className="shadow-lg p-3 mb-5 bg-white rounded form-div">
          <p>Sign In</p>
          <LogInForm
            handleChange={handleChange}
            handleSignInSubmit={handleSignInSubmit}
          />
          <h2><span className="bg-light">OR</span></h2>
          <div className="round-icon">
            <SocialComponent
              facebook={facebook}
              google={google}
              twitter={twitter}
            />
          </div>
        </div>
      </Modal.Body>
    </Modal>
  );
};

LogInModal.propTypes = {
  signInShow: PropTypes.bool.isRequired,
  closeModal: PropTypes.func.isRequired,
  handleChange: PropTypes.func.isRequired,
  handleSignInSubmit: PropTypes.func.isRequired,
  facebook: PropTypes.func.isRequired,
  google: PropTypes.func.isRequired,
  twitter: PropTypes.func.isRequired,
};

export default LogInModal;
