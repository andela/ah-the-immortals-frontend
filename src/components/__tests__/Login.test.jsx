import React from 'react';
import { BrowserRouter as Router } from 'react-router-dom'; 
import { mount } from 'enzyme';
import Login from '../Login';

describe('test dummy login', () => {
  it('returns div element', () => {
    const wrapper = mount(<Router><Login /></Router>);
    expect(wrapper).toMatchSnapshot();
  });
});
