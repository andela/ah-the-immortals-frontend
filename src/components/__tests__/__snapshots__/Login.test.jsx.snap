// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`test dummy login returns div element 1`] = `ReactWrapper {}`;
